### PR TITLE
build: add Github action workflow for publishing release

### LINKED ISSUE #878: Publish release notes automatically with github actions
When cutting new releases, the following commands could be used to generate the release notes:

```sh
GO111MODULE=on go install k8s.io/release/cmd/release-notes@latest

release-notes --repo=cloud-provider-azure \
	--org=kubernetes-sigs \
	--branch=release-1.1 \
	--start-sha=$(git rev-parse v1.1.1) \
	--end-sha=$(git rev-parse v1.1.2) \
	--output=v1.1.2.md
```

The same commands could be executed in github actions and hence the release notes could be published automatically.

### PR DESCRIPTION
**Pull request by luncj**
#### Which issue(s) this PR fixes:

Fixed #878 #879 

#### Special notes for your reviewer:

HOW-TO:
```
git checkout release-1.23
git tag v1.23.2  # create a new release tag
git push upstream v1.23.2  # push tag to upstream
```

example:
- action: https://github.com/luncj/cloud-provider-azure/actions/runs/1712322173
- release: https://github.com/luncj/cloud-provider-azure/releases/tag/v1.23.2

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        uses: actions/download-artifact@v2
+        with:
+          path: ./artifacts
+      - name: Publish release
+        uses: softprops/action-gh-release@v1
+        with:
+          body_path: release-note.md
+          files: |
+            ./artifacts/cloud-node-manager-*-*/*
+            ./artifacts/cloud-controller-manager-*-*/*
```
Comment (MartinForReal) [inline .github/workflows/release.yaml:142]: I'm wondering if building docker images will be a more feasible option.

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+            arch: arm
+          - os: linux
+            arch: arm64
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup Golang
+        uses: actions/setup-go@v2
+        with:
+          go-version: '1.17'
```
Comment (nilo19) [inline .github/workflows/release.yaml:29]: Will this golang version be compatible with release-1.0 and release-1.1?

File: .github/workflows/release.yaml (hunk 1 above)
Comment (luncj) [reply .github/workflows/release.yaml:146]: please take a look at https://github.com/kubernetes-sigs/cloud-provider-azure/issues/879. 
Docker images are available in [MCR](https://mcr.microsoft.com/v2/oss/kubernetes/azure-cloud-node-manager/tags/list), and this PR provides binary as another option.

File: .github/workflows/release.yaml (hunk 2 above)
Comment (luncj) [reply .github/workflows/release.yaml]: fixed. It will load the GO version from `go.mod` dynamically.

File: .github/workflows/release.yaml (hunk 1 above)
Comment (feiskyer) [reply .github/workflows/release.yaml:146]: yea, we only need binaries here. Images would be pushed to MCR

**Review by feiskyer — APPROVED** [review]
Thanks a lot for adding the automations

/lgtm
/approve

**Comment by nilo19** [comment]
manually merged because there is only manifest change

**Pull request merged**